pydicom/dataset.py

```diff
@@ -355,78 +362,164 @@ def __iter__(self):
         for tag in taglist:
             yield self[tag]
 
+    def _is_supported_transfer_syntax(self):
+        # XXX uses file_meta here, should really only be thus for FileDataset
+        return self.file_meta.TransferSyntaxUID in NotCompressedPixelTransferSyntaxes
+
     def _pixel_data_numpy(self):
         """Return a NumPy array of the pixel data if NumPy is available.
+        Falls back to GDCM in case of unsupported transfer syntaxes.
 
         Raises
         ------
         TypeError
-            If there is no pixel data or not a NumPy data type
+            If there is no pixel data or not a supported data type
         ImportError
-            If NumPy isn't found.
+            If NumPy isn't found, or in the case of fallback, if GDCM isn't found.
 
         Returns
         -------
         NumPy array
         """
-        if 'PixelData' not in self:
-            raise TypeError("No pixel data found in this dataset.")
-
         if not have_numpy:
             msg = "The Numpy package is required to use pixel_array, and numpy could not be imported.\n"
             raise ImportError(msg)
-
-        # determine the type used for the array
-        need_byteswap = (self.is_little_endian != sys_is_little_endian)
-
-        # Make NumPy format code, e.g. "uint16", "int32" etc
-        # from two pieces of info:
-        #    self.PixelRepresentation -- 0 for unsigned, 1 for signed;
-        #    self.BitsAllocated -- 8, 16, or 32
-        format_str = '%sint%d' % (('u', '')[self.PixelRepresentation],
-                                  self.BitsAllocated)
-        try:
-            numpy_format = numpy.dtype(format_str)
-        except TypeError:
-            msg = ("Data type not understood by NumPy: "
-                   "format='%s', PixelRepresentation=%d, BitsAllocated=%d")
-            raise TypeError(msg % (numpy_format, self.PixelRepresentation,
-                            self.BitsAllocated))
-
-        # Have correct Numpy format, so create the NumPy array
-        arr = numpy.fromstring(self.PixelData, numpy_format)
-
-        # XXX byte swap - may later handle this in read_file!!?
-        if need_byteswap:
-            arr.byteswap(True)  # True means swap in-place, don't make a new copy
-        # Note the following reshape operations return a new *view* onto arr, but don't copy the data
+        
+        # result container
+        pixel_array = None
+        
+        # FIXME: should we always use GDCM if it is available? Or only as a fallback?
+        if self._is_supported_transfer_syntax(): # and not have_gdcm
```

Comment:
If we uncomment the end of line 392, GDCM will be used first if it is available. If we leave it commented out, it will only be used as a fallback.

Main benefit of using GDCM first: GDCM supports LUT's, pydicom does not.